android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/home/HomeViewModel.kt

```diff
@@ -24,24 +26,63 @@ class HomeViewModel(
     private val _avatarUrlString = MutableStateFlow("")
     val avatarUrlString: StateFlow<String> = _avatarUrlString.asStateFlow()
 
+    private val _surveysUiModel: MutableStateFlow<HomeSurveysUiModel?> = MutableStateFlow(null)
+    val surveysUiModel: StateFlow<HomeSurveysUiModel?> = _surveysUiModel.asStateFlow()
+
+    private lateinit var surveys: List<Survey>
+    private var currentSurveyIndex: Int = 0
+
     init {
         _currentDate.value = dateTimeFormatter.getFormattedString(
             dateTime.today(),
             DateFormat.WeekDayMonthDay
         ).uppercase()
     }
 
-    fun loadProfile() {
+    fun loadProfileAndSurveys() {
         showLoading()
         viewModelScope.launch {
-            getProfileUseCase()
+            val getProfile = getProfileUseCase()
+            // TODO: Improve this when we have paging implementation
+            val getSurveys = getSurveysUseCase(1, 5)
+            getProfile
+                .combine(getSurveys) { profile, surveys -> Pair(profile, surveys) }
                 .catch { e ->
                     showError(e.message)
                 }
                 .collect {
-                    _avatarUrlString.value = it.avatarUrl
+                    _avatarUrlString.value = it.first.avatarUrl
+                    surveys = it.second
```

Comment:
IMO, the `surveys` data is "raw" data. The view model should contain that one for further use (e.g., update logic like `currentSurveyIndex < surveys.size - 1`). After converting to `UiModel`, the `UiModel` should be used for the View (UI). For logic in the view model, we should use the "raw" data. What do you think?

Reply:
@luongvo I did this approach on the UI task 🤪

Reply:
@suho Yeah, sr if I made you confused here. I think you're mentioning [this comment](https://github.com/suho/kmm-ic/pull/106#discussion_r1032189713), right? I think it's because we don't have a full integration yet to review the logic here. 

Sending a list of surveys would make sense in case we will bind them into a view instead of handing them manually. 🙏

Reply:
Fixed in https://github.com/suho/kmm-ic/pull/107/commits/beae7674d690e5854ce56b7a98b4577a2dec5885